deepchecks/base/dataset.py

```diff
@@ -420,19 +420,25 @@ def copy(self: TDataset, new_data: pd.DataFrame) -> TDataset:
                    convert_datetime=False, max_categorical_ratio=self._max_categorical_ratio,
                    max_categories=self._max_categories, label_type=self.label_type)
 
-    def sample(self: TDataset, n_samples: int, replace: bool = False, random_state: t.Optional[int] = None) -> TDataset:
+    def sample(self: TDataset, n_samples: int, replace: bool = False, random_state: t.Optional[int] = None,
+               drop_na_label: bool = True) -> TDataset:
```

Comment:
This should be true only when sampling in metric related checks, not in all checks